File: Java/Organism.java
import java.awt.Color;
public abstract class Organism {
  private final char symbol;
  private Point position;
  private int age;
  private int strength;
  private final int initiative;
  private boolean canAction;
  private final Color color;
  private final String name;
  final World world;

  Organism(World world, Point position, char symbol, int strength, int initiative, Color color, String name) {
    this.world = world;
    this.position = position;
    this.symbol = symbol;
    this.strength = strength;
    this.initiative = initiative;
    this.color = color;
    this.age = 0;
    this.canAction = false;
    this.name = name;
  }

  // Returns false if organism frees the field and true if it keeps its place
  protected abstract boolean Collision(Organism other);
  public abstract void Action();

  public final char GetSymbol() {
    return symbol;
  }

  final String GetName() { return name; }

  public final Color GetColor() {
    return color;
  }

  public final Point GetPosition() {
    return position;
  }
  public final void SetPosition(Point position) {
    this.position = position;
  }

  public final int GetAge() {
    return age;
  }
  public final void IncrementAge() {
    age++;
  }

  public final int GetStrength() {
    return strength;
  }
  public final void SetStrength(int strength) {
    this.strength = strength;
  }

  public final int GetInitiative() {
    return initiative;
  }

  final boolean CanAction() {
    return canAction;
  }
  public final void SetCanAction(boolean canAction) {
    this.canAction = canAction;
  }

  final void Die() {
    this.world.RemoveOrganism(this);
  }

  protected abstract Organism Clone(Point position);
}

File: C++/Organism.cpp
#include "CONSTANTS.h"

#include "World.h"
#include "Organism.h"
#include <iostream>

Organism::Organism(World &world, const Point &position, const char &symbol, const int &strength, const int &initiative, const std::string &color, const std::string &name) :
world(world), position(position), symbol(symbol), strength(strength), initiative(initiative), color(color), name(name) {canAction = false; age = 0;}

Point Organism::GetPosition() const {
  return position;
}

void Organism::SetPosition(const int &x, const int &y) {
  position.SetX(x);
  position.SetY(y); 
}

int Organism::GetAge() const {
  return age;
}

void Organism::IncrementAge() {
  age++;
}

int Organism::GetInitiative() const {
  return initiative;
}

int Organism::GetStrength() const {
  return strength;
}


void Organism::SetStrength(const int &strength) {
  this->strength = strength;
}

char Organism::GetSymbol() const {
  return symbol;
}

std::string Organism::GetName() const {
  return name;
}

void Organism::Draw() {
  std::cout << color << symbol << RESET_COLOR;
}

void Organism::Die() {
  this->world.RemoveOrganism(this);
}

bool Organism::CanAction() const {
  return canAction;
}

void Organism::SetCanAction(const bool &canAction) {
  this->canAction = canAction;
}